### PR TITLE
Adding System.Device.Pwm

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Adding System.Device.Pwm in the context of API alignment with .NET IoT.

## Motivation and Context

- Adding System.Device.Pwm in the context of API alignment with .NET IoT.
- System.Device.Pwm repository needs to go for PR as well
- Samples will have to be added as well

## How Has This Been Tested?

Need to be tested for STM32 device. Tested on ESP32.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Ellerbach** [comment]
@AndersTh for your visibility as well.

File: CMake/Modules/FindNF_NativeAssemblies.cmake [hunk 1]
```diff
@@ -26,6 +26,7 @@ option(API_System.Device.I2c                    "option for System.Device.I2c AP
 option(API_Windows.Devices.Gpio                 "option for Windows.Devices.Gpio API")
 option(API_Windows.Devices.I2c                  "option for Windows.Devices.I2c API")
 option(API_Windows.Devices.Pwm                  "option for Windows.Devices.Pwm API")
+option(API_Windows.Devices.Pwm                  "option for Windows.Devices.Pwm API")
```
Comment (josesimoes) [inline CMake/Modules/FindNF_NativeAssemblies.cmake:29]: guess this should be System.Device.Pwm...
```suggestion
option(API_System.Device.Pwm                  "option for System.Device.Pwm API")
```

File: targets/ChibiOS/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+#ifndef _SYS_DEV_PWM_NATIVE_TARGET_H_
+#define _SYS_DEV_PWM_NATIVE_TARGET_H_
+
+#include <sys_dev_pwm_native.h>
+
+enum PwmPulsePolarity
```
Comment (josesimoes) [inline targets/ChibiOS/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h:11]: This looks like a duplicate from typedef enum __nfpack PwmPulsePolarity @ src/System.Device.Pwm/sys_dev_pwm_native.h

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        // Work out the duty Cycle for the current duty resolution
+        dutyCycle = CalculateDuty(timerId, dutyCycle, polarity);
+
+        ledc_channel_config_t ledc_conf{pinNumber, mode, channel, LEDC_INTR_DISABLE, timer_sel, dutyCycle, 0};
+
+        // Configure Channel which will also start it
+        IDF_ERROR(ledc_channel_config(&ledc_conf));
+
+        // Because it is started from the configure we optionally stop it and set idle level based on polarity
+        if (noStart)
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp:128]: Open & close brackets please

File: targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+#ifndef _SYS_DEV_PWM_NATIVE_TARGET_H_
+#define _SYS_DEV_PWM_NATIVE_TARGET_H_
+
+#include <sys_dev_pwm_native.h>
+
+enum PwmPulsePolarity
```
Comment (josesimoes) [inline targets/FreeRTOS_ESP32/ESP32_WROOM_32/nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h:11]: Same here...

File: targets/TI_SimpleLink/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+#ifndef _SYS_DEV_PWM_NATIVE_TARGET_H_
+#define _SYS_DEV_PWM_NATIVE_TARGET_H_
+
+#include <sys_dev_pwm_native.h>
+
+enum PwmPulsePolarity
```
Comment (josesimoes) [inline targets/TI_SimpleLink/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h:11]: Same here

File: targets/ChibiOS/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h (hunk 2 above)
Comment (Ellerbach) [reply targets/ChibiOS/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_target.h:11]: yes, will remove this one. Good catch

**Comment by josesimoes** [comment]
@Ellerbach tested on STM32F769. Please test on ESP32 😉

**Comment by josesimoes** [comment]
@Ellerbach build OK now. Please review the changes to ESP32 implementation and maybe perform some testing...

**Comment by Ellerbach** [comment]
Great team work on this one, @AndersTh and @josesimoes really happy to see this API merged!

**Review by josesimoes — APPROVED** [review]
Awesome work here @Ellerbach and @AndersTh!

**Pull request merged**